### ISSUE TITLE
Breaks due of Contao 4.5 / PHP 7.1 type declarations

### ISSUE THREAD

**Issue by richardhj** [contao-community-alliance/dc-general#397]
## My environment is:

| Key                                 | Value    | Comments                         |
| ----------------------------------- | ---------| ---------------------------------|
| PHP version:                        |     7.1.6     |                                  |
| Contao version:                     |   4.5.0      |                                  |
| DC_General version:                 |      2.1@dev    |                                  |

## Issue description

I made my first installation with Contao 4.5 and MetaModels, and therefore with DCG.
We have some breaks:
```
Fatal error: Declaration of ContaoCommunityAlliance\DcGeneral\Contao\Picker\AbstractAwarePickerProvider::setTokenStorage(Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface $tokenStorage) must be compatible with Contao\CoreBundle\Picker\AbstractPickerProvider::setTokenStorage(Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface $tokenStorage): void in /Users/richard/Sites/sjralfeld.dev/vendor/contao-community-alliance/dc-general/src/Contao/Picker/AbstractAwarePickerProvider.php on line 118
```
The culprit is obvious here: [AbstractPickerProvider::setTokenStorage](https://github.com/contao/core-bundle/blob/62db4248470ff911d337a79e3cb47d044fab63b5/src/Picker/AbstractPickerProvider.php#L94) (part of the contao/core-bundle) now has `:void` as return type declaration since the 4.5 strict-types release.

How to handle this? We cannot provide a version 2.1 for Contao 4.4 and a version 2.2 for Contao >=4.5 at the same time.

@contao-community-alliance/members

**Comment by discordier** [comment]
Can't we switch to use the  `PickerProviderInterface` instead of `AbstractPickerProvider` and therefore omit using the breaking parent class?

Then we could be compatible with 4.4. and 4.5 again.

//cc @baumannsven

**Comment by richardhj** [comment]
Please review my PR.

**Comment by richardhj** [comment]
Fixed in deb0716687858ec8290e3ed5a9d1c19bc02b8c68.
For now, this has been the only Contao 4.5 incompatibility.
